### PR TITLE
Add podcastByGuid

### PR DESCRIPTION
**Pull request by iwootten**
Adds podcast lookup through using it's guid.

### CODE REVIEW

File: tests/test_podcast_lookup.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 feedUrl = "http://feed.thisamericanlife.org/talpodcast"
 feedId = 522613
 itunesId = 201671138
+guid = "f1ebeaa1-bc5a-534f-8528-0738ae374d55"
```
Comment (SarvagyaVaish) [inline tests/test_podcast_lookup.py:12]: Could we call this `podcastGuid` perhaps?

File: tests/test_podcast_lookup.py [hunk 2]
```diff
@@ -33,3 +34,10 @@ def test_podcast_lookup_by_byitunesid():
 
     results = index.podcastByItunesId(itunesId)
     assert results["feed"]["title"] == podcastTitle, "Did not find the right podcast when doing lookup by Itunes ID"
+
+def test_podcast_lookup_by_byguid():
+    config = podcastindex.get_config_from_env()
+    index = podcastindex.init(config)
+
+    results = index.podcastByGuid(guid)
+    assert results["feed"]["title"] == podcastTitle, "Did not find the right podcast when doing lookup by guid"
```
Comment (SarvagyaVaish) [inline tests/test_podcast_lookup.py:43]: Thanks for adding a test proactively :)

**Review by SarvagyaVaish — CHANGES_REQUESTED** [review]
Thanks so much for contributing to the repo! Great addition :) Requesting one small change for clarity. Will approve, merge, and release a new version in PyPi after that.

File: tests/test_podcast_lookup.py (hunk 1 above)
Comment (SarvagyaVaish) [reply tests/test_podcast_lookup.py:12]: Ahh nvm. This is already in the podcast specific test file.

**Comment by SarvagyaVaish** [comment]
Thank you so much for your contribution. A new version with your contribution is up! 
https://pypi.org/project/python-podcastindex/#history

**Comment by iwootten** [comment]
Thanks for reviewing this so quickly!

**Pull request merged**